edkrepo/common/common_repo_functions.py

```diff
@@ -638,17 +638,23 @@ def checkout(combination, global_manifest_path, verbose=False, override=False, l
 def get_latest_sha(repo, branch, remote_or_url='origin'):
     if repo is None:
         try:
-            ls_remote_output = subprocess.run('git ls-remote {} {}'.format(remote_or_url, branch),
-                                               stdout=subprocess.PIPE, universal_newlines=True, shell=True)
-            latest_sha = ls_remote_output.stdout.split()[0]
+            ls_remote_output = subprocess.run('git ls-remote {} refs/heads/{}'.format(remote_or_url, branch),

```

Comment:
Just a comment, to give more options for future changes, can we try below:

REMOTE_PREFIX = get_remote_prefix(env)

def get_remote_prefix(env = None):
    if not env:
        return "refs/heads/"
    elif:
        return "refs/for/"
    ## and other conditions

Reply:
I think for now it is fine to leave it as refs/heads/, as it is how Git organizes refs by default. "refs/for/" seems to be a Gerrit-specific syntax so we probably don't need to include it here.